File: src/event/impl/template/TemplateEditEvent.php
<?php

namespace pocketcloud\cloud\event\impl\template;

use pocketcloud\cloud\template\Template;

class TemplateEditEvent extends TemplateEvent {

    public function __construct(
        Template $template,
        protected readonly ?bool $lobby,
        protected readonly ?bool $maintenance,
        protected readonly ?bool $static,
        protected readonly ?int $maxPlayerCount,
        protected readonly ?int $minServerCount,
        protected readonly ?int $maxServerCount,
        protected readonly ?float $startNewPercentage,
        protected readonly ?bool $autoStart
    ) {
        parent::__construct($template);
    }

    public function getLobby(): ?bool {
        return $this->lobby;
    }

    public function getMaintenance(): ?bool {
        return $this->maintenance;
    }

    public function getStatic(): ?bool {
        return $this->static;
    }

    public function getMaxPlayerCount(): ?int {
        return $this->maxPlayerCount;
    }

    public function getMinServerCount(): ?int {
        return $this->minServerCount;
    }

    public function getMaxServerCount(): ?int {
        return $this->maxServerCount;
    }

    public function getStartNewPercentage(): ?float {
        return $this->startNewPercentage;
    }

    public function getAutoStart(): ?bool {
        return $this->autoStart;
    }
}

File: src/main/java/de/pocketcloud/cloud/event/impl/template/TemplateEditEvent.java
package de.pocketcloud.cloud.event.impl.template;

import de.pocketcloud.cloud.event.Cancelable;
import de.pocketcloud.cloud.template.Template;
import de.pocketcloud.cloud.template.util.TemplateEditData;
import lombok.Getter;

public final class TemplateEditEvent extends TemplateEvent implements Cancelable {

    @Getter
    private final TemplateEditData editData;

    public TemplateEditEvent(Template template, TemplateEditData editData) {
        super(template);
        this.editData = editData;
    }
}